test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java

```diff
@@ -0,0 +1,212 @@
+                EventWriterConfig.builder().build());
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        for (int i = 0; i < numMessages; i++) {
+            futures.add(writer.writeEvent(message));
+        }
+        Futures.allOf(futures).join();
+        log.info("Wrote {} message(s) to the stream {}/{}", numMessages, scope, stream);
+    }
+
+    /**
+     * Returns the first message from the specified {@code scope}/{@code stream}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param readerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @param readerGroup the name of the reader group application
+     * @return the event message
+     */
+    public static String readAMessageFromStream(String scope, String stream, ClientConfig readerClientConfig,
+                                                String readerGroup) {
+        List<String> messages = readDataFromStream(scope, stream, 1, readerClientConfig, readerGroup);
+        if (messages == null || messages.size() == 0) {
+            return null;
+        } else {
+            return messages.get(0);
+        }
+    }
+
+    /**
+     * Returns the specified number of messages from the specified {@code scope}/{@code stream}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param numMessages the number of event messages to write
+     * @param readerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @param readerGroup the name of the reader group application
+     * @return the event messages
+     * @throws NullPointerException if {@code scope} or {@code stream} or {@writerClientConfig} is null
+     * @throws IllegalArgumentException if {@code numMessages} < 1
+     * @throws RuntimeException if any exception is thrown by the client
+     */
+    public static List<String> readDataFromStream(@NonNull String scope, @NonNull String stream, int numMessages,
+                                                  @NonNull ClientConfig readerClientConfig, @NonNull String readerGroup) {
+        Preconditions.checkArgument(numMessages > 0);
+
+        @Cleanup
+        EventStreamClientFactory readerClientFactory = EventStreamClientFactory.withScope(scope, readerClientConfig);
+        log.debug("Created the readerClientFactory");
+
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(scope, stream))
+                .disableAutomaticCheckpoints()
+                .build();
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, readerClientConfig);
+        readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+        log.debug("Created reader group with name {}", readerGroup);
+
+        @Cleanup
+        EventStreamReader<String> reader = readerClientFactory.createReader(
```

Comment:
Since the reader group is not deleted invoking `readDataFromStream(..)` again would cause the method to read the next events from the stream. So we should document this behaviour or perhaps invoke delete reader group or change the name of the method to convey this behaviour perhaps as `readNextEvents(..)`

Reply:
Sure, I've renamed the methods, so that they better represent the behavior that these methods exhibit.